lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -226,7 +204,21 @@ def __get_k8s_job_spec_template(self, ajs):
         (see pod selector) and an appropriate restart policy."""
         k8s_spec_template = {
             "metadata": {
-                "labels": {"app": self.__produce_unique_k8s_job_name(ajs.job_wrapper.get_id_tag())[:-5]}
+                "labels": {
+                    "app.kubernetes.io/name": ajs.job_wrapper.tool.old_id,
+                    "app.kubernetes.io/instance": self.__produce_k8s_job_prefix(),
+                    "app.kubernetes.io/version": ajs.job_wrapper.tool.version,
+                    "app.kubernetes.io/component": "tool",
+                    "app.kubernetes.io/part-of": "galaxy",
+                    "app.kubernetes.io/managed-by": "galaxy",
+                    "app.galaxyproject.org/job_id": ajs.job_wrapper.get_id_tag(),
+                    "app.galaxyproject.org/instance": self._galaxy_instance_id or "",
+                    "app.galaxyproject.org/handler": self.app.config.server_name,
+                    "app.galaxyproject.org/destination": ajs.job_wrapper.job_destination.id,
+                },
+                "annotations": {
+                    "app.galaxyproject.org/tool_id": ajs.job_wrapper.tool.id,
```

Comment:
@mvdbeek I just noticed that this is is still referring to tool.id?

Reply:
From https://github.com/galaxyproject/galaxy/pull/10407#discussion_r504816337 I understood this only applies to labels. Is that not correct ?

Reply:
Aah yes, that makes sense. Label selectors need to be dns compliant/url friendly.